client/src/components/HELData/HELData.jsx

```diff
@@ -0,0 +1,142 @@
+import React, { useState } from "react";
+import { Chart } from "react-charts";
+import { searchHEL } from "../../utils/helData";
+import Search from "../Search";
+import ItemList from "../ItemList";
+import * as TimeTableData from "../../Timetable.json";
+import { useGetData } from "use-axios-react";
+import axios from "axios";
+const course = JSON.parse(JSON.stringify(TimeTableData)).default;
+
+let axes = [];
+let courseData = [];
+let humCourses = Object.keys(course)
+    .filter(
+        code =>
+            code.startsWith("GS") ||
+            code.startsWith("HSS") ||
+            code.startsWith("BITS F214") ||
+            code.startsWith("BITS F385") ||
+            code.startsWith("BITS F399")
+    )
+    .reduce((res, key) => ((res[key] = course[key]), res), {});
+
+
+
+const HELData = () => {
+    
+    const [studentData,setStudentData] = useState([]);
+    let result = ""; 
+
+    const HELstats = (e) => {
+        let et = e.target.innerHTML.toLowerCase();
+        let event = et.split(" ");
+        event = event[0] + " " + event[1];
+        console.log(event);
+        let finished = false;
+        
+        
+        try {
+             axios.get("/api/helData/searchHEL", {
+                params: {
+                    "courseData": event
+                }
+            }
+            ).then(res => {
+                result = res.data.studentsInterestedInAllSlots;
+            })
+        }
+        catch (err) {
+            console.log("DB RETRIEVAL ERROR:", err);
+        }
```

Comment:
```suggestion
    const [studentData, setStudentData] = useState({
    courseStats: []
  });
  const { courseStats } = studentData;
    let result = ""; 

    const HELstats = (e) => {
        e.preventDefault();
    let et = e.target.innerHTML.toLowerCase();
    let event = et.split(" ");
    event = event[0] + " " + event[1];
    console.log(event);
    let finished = false;

    try {
      axios
        .get("/api/helData/searchHEL", {
          params: {
            courseData: event
          }
        })
        .then(res => {
          console.log("here");
          resp = true;
          result = res.data.studentsInterestedInAllSlots;
          console.log(result);
          setStudentData({ ...studentData, courseStats: result });
        });
        catch (err) {
            console.log("DB RETRIEVAL ERROR:", err);
        }
```

This is working as expected, I think.

Reply:
Also you can have your endpoint as `api/helData/searchHEL/:name` and call it as `.get(`\`api/helData/searchHEL/${name}`)
With backticks and not single quotes. Makes the code shorter.
Both work, just a suggestion

Reply:
so what you're saying is that I pass the param in the url?

Reply:
You can, yes. It will make for more readable code.